bin/strata-sequencer-client/src/duty_executor.rs

```diff
@@ -67,16 +67,18 @@ where
 async fn handle_duty<R>(
     rpc: Arc<R>,
     duty: Duty,
-    idata: IdentityData,
+    idata: Arc<IdentityData>,
     failed_duties_tx: mpsc::Sender<DutyId>,
 ) where
     R: StrataSequencerApiClient + Send + Sync,
 {
     let duty_id = duty.id();
     debug!(%duty_id, ?duty, "handle_duty");
     let duty_result = match duty.clone() {
-        Duty::SignBlock(duty) => handle_sign_block_duty(rpc, duty, duty_id, idata).await,
-        Duty::CommitBatch(duty) => handle_commit_batch_duty(rpc, duty, duty_id, idata).await,
+        Duty::SignBlock(duty) => handle_sign_block_duty(rpc, duty, duty_id, idata.clone()).await,
+        Duty::CommitBatch(duty) => {
+            handle_commit_batch_duty(rpc, duty, duty_id, idata.clone()).await
```

Comment:
*Possibly* these calls should actually take `&IdentityData` to avoid having to clone the arc.  I am not sure if they go on to move it into another thread (so actually *need* to own it) so I might be wrong.